Update CUDA architectures: remove sm_61/62/70/72 (dropped in CUDA 13), add sm_89/90.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,14 +2,14 @@
 project(bmtrain)
 enable_language(C)
 enable_language(CXX)
-set(CMAKE_CUDA_ARCHITECTURES "61;62;70;72;75;80")
+set(CMAKE_CUDA_ARCHITECTURES "75;80;86;89;90")
 enable_language(CUDA)
 set(CMAKE_CXX_STANDARD 14)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 set(CMAKE_CUDA_STANDARD 14)
 set(CMAKE_CUDA_STANDARD_REQUIRED True)

-set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_61,code=sm_61 -gencode=arch=compute_62,code=sm_62 -gencode=arch=compute_70,code=sm_70 -gencode=arch=compute_72,code=sm_72 -gencode=arch=compute_75,code=sm_75 -gencode=arch=compute_80,code=sm_80")
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_75,code=sm_75 -gencode=arch=compute_80,code=sm_80 -gencode=arch=compute_89,code=sm_89 -gencode=arch=compute_90,code=sm_90")

 if(NOT DEFINED ENV{BUILD_DOCKER_ENV} OR "$ENV{BUILD_DOCKER_ENV}" STREQUAL "0")
     set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -gencode=arch=compute_86,code=sm_86")
@@ -54,7 +54,7 @@
 target_compile_definitions(C
     PRIVATE VERSION_INFO=${EXAMPLE_VERSION_INFO})

-set_target_properties(C PROPERTIES CUDA_ARCHITECTURES "61;62;70;72;75;80")
+set_target_properties(C PROPERTIES CUDA_ARCHITECTURES "75;80;86;89;90")

 target_include_directories(C
     PRIVATE "csrc/include"
